pageserver/src/layered_repository/delta_layer.rs

```diff
@@ -316,6 +316,18 @@ impl Layer for DeltaLayer {
         }
     }
 
+    fn key_iter<'a>(&'a self) -> Box<dyn Iterator<Item = (Key, Lsn, u64)> + 'a> {
+        let inner = match self.load() {
+            Ok(inner) => inner,
+            Err(e) => panic!("Failed to load a delta layer: {e:?}"),
```

Comment:
Hmm.
May be I am missing something...
First of all panic! is called not only in `key_iter` (which I have added), but also in exited iter().
The only difference is that `key_iter()` calls panic not only when `self.load()` is failed, but also
when `DeltaKeyLayer::new` is failed. Please notice that unlike `DeltaValueIter::new`, this function is not loaded BLOB fro the disk. So it can fail only if ` tree_reader.visit` is failed.

In any case - `Iterator::next` should return `Option`, not `Result`. So replacing `panic!` with `bail!` is not possible.
`DeltaValueIter::next` may return error as Option result. But key_iter is not extracting BLOBs, so `DeltsKeyIter::next` need not  propagate error.